src/openfl/display/Stage.hx

```diff
@@ -682,13 +677,20 @@ class Stage extends DisplayObjectContainer implements IModule {
 	public function onRenderContextLost (renderer:Renderer):Void {
 		
 		__renderer = null;
+		var stage3D = stage3Ds[0];
+		if (stage3D.context3D != null) {
+			stage3D.__contextRequested = true; // because we want to dispatch `context3DCreate`
+		}
+		stage3D.context3D = null;
 		
 	}
 	
 	
 	public function onRenderContextRestored (renderer:Renderer, context:RenderContext):Void {
 		
+		GLTextureBase.reset();
```

Comment:
Maybe move this to `onRenderContextLost` to have all the resetting done on one function?

Reply:
It was there originally, but then there was a case with texture formats being requested between lost and restore handlers, breaking it again. I believe it was happening when async loading and texture creation (and thus supported format query) happens when the context is lost. So this way it's a bit less fragile.